src/daft-plan/src/logical_ops/join.rs

```diff
@@ -167,6 +169,35 @@ impl Join {
         }
     }
 
+    fn deduplicate_exprs(exprs: Vec<Arc<Expr>>) -> Vec<Arc<Expr>> {
+        let mut counts: HashMap<Arc<Expr>, usize> = HashMap::new();
+
+        exprs
+            .into_iter()
+            .map(|expr| {
+                let count = counts.entry(expr.clone()).or_insert(0);
+                *count += 1;
+
+                if *count == 1 {
+                    expr // First occurrence, return the original expression
+                } else {
+                    let unique_id = Uuid::new_v4();
+                    expr.alias(format!("{}", unique_id)) // Append count for duplicates
+                }
+            })
+            .collect()
+    }
+
+    fn process_expressions(
+        left_on: Vec<Arc<Expr>>,
+        right_on: Vec<Arc<Expr>>,
+    ) -> (Vec<Arc<Expr>>, Vec<Arc<Expr>>) {
+        let unique_left_on = Self::deduplicate_exprs(left_on);
+        let unique_right_on = Self::deduplicate_exprs(right_on);
+
+        (unique_left_on, unique_right_on)
+    }
```

Comment:
I would suggest just calling `deduplicate_exprs` twice instead of making a function that wraps that.

Reply:
Indeed, it doesn't do much than calling another method! I will change this.